core/internal/filetransfer/file_transfer_gcs.go

```diff
@@ -25,22 +37,33 @@ type GCSFileTransfer struct {
 	ctx context.Context
 }
 
+const maxWorkers int = 1000
+
 var ErrObjectIsDirectory = errors.New("object is a directory and cannot be downloaded")
 
 // NewGCSFileTransfer creates a new fileTransfer
 func NewGCSFileTransfer(
 	client GCSClient,
 	logger *observability.CoreLogger,
 	fileTransferStats FileTransferStats,
-) *GCSFileTransfer {
+) (*GCSFileTransfer, error) {
 	ctx := context.Background()
+	if client == nil {
+		var err error
+		client, err = storage.NewClient(ctx)
```

Comment:
nit: you could probably inline `ctx` here

```go
storage.NewClient(context.Background())
```

Reply:
ended up keeping this just because we also add it to the `GCSFileTransfer` struct